### PR TITLE
refactor: scaffold css framework (part 5)

### PR DESCRIPTION
**Pull request by carlosescri**

### CODE REVIEW

**Comment by carlosescri** [comment]
Using the widest version of lit-element and lit-html as recommended in lit-element docs:

image

**Comment by carlosescri** [comment]
image

**Comment by carlosescri** [comment]
image

File: packages/core/src/context.ts [hunk 1]
```diff
@@ -446,6 +446,7 @@ export class Context extends Observable {
     this._requestManager.reset();
     this._query = this.newQuery();
     this._response = null;
+    this.notify(ContextTopic.SearchRequest, { request: null, context: this });
```
Comment (carlosescri) [inline packages/core/src/context.ts:449]: Allow input widgets to reset too…

File: packages/core/src/history.ts [hunk 2]
```diff
@@ -81,8 +91,12 @@ export class SearchHistory {
...
    */
   public pushState(context: Context, query: Query): void {
-    const hash: string = this._serialize(context, query);
-    this._updateState(hash);
+    if (query) {
+      const hash: string = this._serialize(context, query);
+      this._updateState(hash);
+    } else {
+      this.clearState();
+    }
```
Comment (carlosescri) [inline packages/core/src/history.ts:99]: Clear history on reset.

File: packages/results/src/results.ts [hunk 3]
```diff
@@ -404,13 +404,11 @@ export class ResultsWidget extends ReactiveWidget {
...
-          : repeat(
-              this.results,
-              (result) => result.dfid,
-              (data) => renderTemplate(data)
-            )}
+        ${repeat(
+          this.results,
+          (result) => result.dfid,
+          (data) => renderTemplate(data)
+        )}
```
Comment (carlosescri) [inline packages/results/src/results.ts:411]: Bugfix and less code.

File: packages/results/src/results.ts [hunk 4]
```diff
@@ -553,7 +551,7 @@ export class ResultsWidget extends ReactiveWidget {
     this._total = null;
     this._currentPage = 0;
     this._lastPage = 0;
-    this.results = null;
+    this.results = [];
```
Comment (carlosescri) [inline packages/results/src/results.ts:554]: Bugfix.

File: package.json [hunk 5]
```diff
@@ -40,7 +40,7 @@
     "karma": "^5.0.5",
     "karma-chai": "^0.1.0",
     "lint-staged": "^9.4.2",
-    "lit-html": "^1.2.1",
+    "lit-html": "^1.0.0",
```
Comment (sonic182) [inline package.json:43]: problems with version 1.2.1?

File: packages/banner/package.json [hunk 6]
```diff
@@ -17,7 +17,8 @@
   },
   "dependencies": {
     "@dfwc/core": "^0.0.0",
-    "lit-element": "^2.3.1",
+    "lit-element": "^2.0.0",
```
Comment (sonic182) [inline packages/banner/package.json:20]: problems with 2.3.1?

File: packages/results/src/results.ts (hunk 4 above)
Comment (sonic182) [reply packages/results/src/results.ts:554]: The property setter should check that results never it's null, if so, set empty array

File: packages/filters/src/tags/color.ts [hunk 7]
```diff
@@ -0,0 +1,32 @@
+import { property, CSSResultArray, html, TemplateResult } from 'lit-element';
+import { contrast } from '@dfwc/core';
+import { TagWidget } from './tag';
+import { styles } from './styles/color';
+
+export class ColorTagWidget extends TagWidget {
```
Comment (sonic182) [inline packages/filters/src/tags/color.ts:6]: how about adding a little explication as a comment?

File: packages/results/src/results.ts (hunk 4 above)
Comment (carlosescri) [reply packages/results/src/results.ts:554]: I don't think so… the component must work as expected and handle reasonable cases. If you pass a non iterable value the component will break as expected…

File: package.json (hunk 5 above)
Comment (carlosescri) [reply package.json:43]: See: https://github.com/doofinder/webcomponents/pull/328#issuecomment-633039479

File: packages/banner/package.json (hunk 6 above)
Comment (carlosescri) [reply packages/banner/package.json:20]: See: https://github.com/doofinder/webcomponents/pull/328#issuecomment-633039479

File: packages/filters/src/tags/color.ts (hunk 7 above)
Comment (carlosescri) [reply packages/filters/src/tags/color.ts:6]: This is one of those components available in v7 and this is a split of the tag component. Documentation will be added later (there's an issue for adding docs and finish components).

**Pull request merged**